### DISCUSSION TITLE
Is it possible in automations to bind to multiple looping blocks?

### DISCUSSION THREAD

**Discussion by sl3d-space** [Budibase/budibase#8927] [Help]
My automation works as follows (image1.png):
1- Query a table and return an array of rows
2- Loop through rows while calling an API and producing a value for each row
3- Update each of the rows in the table with the corresponding result of the API call

The problem I have is in step 3 (image2.png). In order to update a row I need the currently iterated row id, so I bind this loop to steps.1.rows. However, I also want to access the corresponding result from step 2, which it seems I can only do by binding to steps.2.items instead.

If there's an index number available through the loop, I could use it to index into both arrays. Or is there any other way to achieve this?

Thanks for reading

![image2](https://user-images.githubusercontent.com/89758363/205549840-2269a439-fc3a-4bd6-8aef-859e65c68dfe.png)
![image1](https://user-images.githubusercontent.com/89758363/205549845-f4a35165-7f6b-4f46-932b-3ef1aa0c4697.png)

**Comment by melohagan** [comment]
Hey @sl3d-space 

There is no way to have nested loops. Each looped block is run sequentially.

**Reply by sl3d-space** [reply]
Hi @melohagan,

This does not require a nested loop. Here's some pseudocode to explain what I'm trying to do:

```python
rows = Table.Get(query)
for row in rows:
  result = API(row)
  row.update(result)
```
Now here's the issue: in the final looped update block, I bind to the array returned by my api call, but in order to update the row I also need to bind to Query Rows to access the row id. However, a single looped block can only bind to one array. Both are needed to be able to call update, with my API call result, on the correct row, but only one array can be bound at a time. 

Hope this is clear and thanks for responding!

**Reply by melohagan** [reply]
Hey @sl3d-space That makes sense to me now, thank you for the pseudo code!

So given this setup which I believe is similar to yours:
![Screenshot 2022-12-09 at 10 42 11](https://user-images.githubusercontent.com/101575380/206684369-67323b00-a61c-487c-9a56-4c13da04b1da.png)

In your final **Update rows** loop, you'll need to effectively merge the outputs of the *Query Rows* step and the *External Data Connector*. Hopefully the result will provide some sort of id that can be tied to the row that you want to update?

![Screenshot 2022-12-09 at 10 50 43](https://user-images.githubusercontent.com/101575380/206686043-371c4697-7670-4e90-b9b8-1b4abcb5025f.png)

This is untested, but I think your binding would look something like this:
```js
let rows = $("steps.1.rows");
let results = $("steps.2.items");

for (let row of rows) {
	row.result = results.find(res => res.id === row.id)
}

return rows
```
